src/Nethermind/Nethermind.Core/Eip6110Constants.cs

```diff
@@ -0,0 +1,10 @@
+// SPDX-FileCopyrightText: 2024 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+namespace Nethermind.Core;
+
+public static class Eip6110Constants
+{
+    public static readonly Address MainnetDepositContractAddress = new("0x00000000219ab540356cbb839cbe05303d7705fa");
```

Comment:
Shouldn't this go to `MainnetSpecProvider`?

Reply:
`MainnetSpecProvider` is used for tests, it is a class representation of mainnet chainspec. I think we had this convention with constants for a longer time. We have Eip1559Constants, 4844, 4788 and so on. We can discuss if we want to change it, but for sure out of the scope of this PR